inference-engine/src/mkldnn_plugin/nodes/mkldnn_adaptive_pooling.cpp

```diff
@@ -0,0 +1,363 @@
+        } else if (one_of(op->get_type_info(), ngraph::op::v8::AdaptiveMaxPool::type_info)) {
+            auto adaPool = std::dynamic_pointer_cast<ngraph::opset8::AdaptiveMaxPool>(op);
+            if (!adaPool) {
+                errorMessage = "Only opset8 AdaptiveMaxPooling operation is supported";
+                return false;
+            }
+        } else {
+            errorMessage = "Unsupported Adaptive pooling mode";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+MKLDNNAdaPoolingNode::MKLDNNAdaPoolingNode(const std::shared_ptr<ngraph::Node>& op, const mkldnn::engine& eng,
+                                           MKLDNNWeightsSharing::Ptr &cache) : MKLDNNNode(op, eng, cache) {
+    auto maxPoolOp = ngraph::as_type_ptr<ngraph::op::v8::AdaptiveMaxPool>(op);
+    auto avgPoolOp = ngraph::as_type_ptr<ngraph::op::v8::AdaptiveAvgPool>(op);
+    std::string errorMessage;
+    if (maxPoolOp) {
+        algorithm = Algorithm::AdaptivePoolingMax;
+    } else if (avgPoolOp) {
+        algorithm = Algorithm::AdaptivePoolingAvg;
+    }
+    if (isSupportedOperation(op, errorMessage)) {
+        errorPrefix = "AdaPooling layer with name '" + getName() + "' ";
+    } else {
+        IE_THROW(NotImplemented) << errorMessage;
+    }
+}
+
+void MKLDNNAdaPoolingNode::getSupportedDescriptors() {
+    if (!descs.empty())
+        return;
+
+    if (getParentEdges().size() != 2)
+        IE_THROW() << errorPrefix << "has incorrect number of input edges: " << getParentEdges().size();
+    if (getChildEdges().empty())
+        IE_THROW() << errorPrefix << "has incorrect number of output edges: " << getChildEdges().size();
+    if (getChildEdges().size() != (algorithm == AdaptivePoolingMax ? 2 : 1))
+        IE_THROW() << errorPrefix << "has incorrect number of output edges: " << getParentEdges().size();
+
+    inputPrecision = getOriginalInputPrecisionAtPort(0);
+    outputPrecision = getOriginalOutputPrecisionAtPort(0);
+
+    auto parentDims = getParentEdgeAt(0)->getDims();
+    auto childDims = getChildEdgeAt(0)->getDims();
+
+    spatialDimsCount = parentDims.ndims() - 2;
+    if (spatialDimsCount != 1 && spatialDimsCount != 2 && spatialDimsCount != 3) {
+        IE_THROW() << errorPrefix << "doesn't support 0th input with rank: " << getParentEdgeAt(0)->getDims().ndims();
+    }
+
+    if (getParentEdgeAt(1)->getDims().ndims() != 1) {
+        IE_THROW() << errorPrefix << "doesn't support 1st input with rank: " << getParentEdgeAt(1)->getDims().ndims();
+    }
+
+    if (getChildEdgeAt(0)->getDims().ndims() != getParentEdgeAt(0)->getDims().ndims()) {
+        IE_THROW() << errorPrefix << "must keep data rank";
+    }
+}
+
+void MKLDNNAdaPoolingNode::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+    // we supports only floating-point types
+    if ((inputPrecision != Precision::FP32 && inputPrecision != Precision::BF16) ||
```

Comment:
Looks like here we just need to check mayiuse(avx512_core) && (outputPrecision == Precision::BF16 || inputPrecision == Precision::BF16) for BF16 and set FP32 otherwise.

Reply:
Done